Added Diacritic Removal Transformation

vyraun opened the pull request:
This transformation removes the accented characters or diacritics on characters, replacing them with their non-accented versions. This transformation also has a sister filter at https://github.com/GEM-benchmark/NL-Augmenter/pull/272. Thanks.

mmeyer39 commented:
There is a typo in the PR name (tranformation -> transformation) and in the first commit message. Could you fix it?

mmeyer39 commented on transformations/diacritic_removal/README.md:
```diff
@@ -0,0 +1,16 @@
+Author: Vikas Raunak (viraunak@microsoft.com)
+
+## What type of a transformation is this?
+This transformation removes the accented characters or diacritics on characters, and replaces them with their non-accented versions.
```
Could you add an example of an accented character and a diacritic character?

vyraun replied:
added the example of diacritic marks

mmeyer39 commented on transformations/diacritic_removal/README.md:
```diff
@@ -0,0 +1,16 @@
+This transformation removes the accented characters or diacritics on characters, and replaces them with their non-accented versions.
+
+## What tasks does it intend to benefit?
+This can simplify the dataset to be used for any task, especially which rely on constructing the embeddings from scratch, since often these rare characters with diacritics lead to poorly formed embeddings. 
```
There seems to be a word missing in the sentence after `especially`.

vyraun replied:
thanks, made the sentence clear now

mmeyer39 commented on transformations/diacritic_removal/README.md:
```diff
@@ -0,0 +1,16 @@
+This can simplify the dataset to be used for any task, especially which rely on constructing the embeddings from scratch, since often these rare characters with diacritics lead to poorly formed embeddings. 
+
+## Previous Work
+No specific research question is entailed here, except that accented characters might be among the rarest of characters, so it would be inetersting to see how much performance gain one can get by replacing them.
```
There is a typo in `interesting`.

vyraun replied:
thanks, fixed it

mmeyer39 commented on transformations/diacritic_removal/README.md:
```diff
@@ -0,0 +1,16 @@
+No specific research question is entailed here, except that accented characters might be among the rarest of characters, so it would be inetersting to see how much performance gain one can get by replacing them.
+
+## What are the limitations of this transformation?
+The transformation is very simple, it just replaces the accented characters. Howver, one has to be careful to not apply it on chemical formulas or phycial units, since those cases diacritics are super important.
```
There is a typo to `However` and to `physical`.

vyraun replied:
thanks, fixed them

mmeyer39 commented on transformations/diacritic_removal/__init__.py:
```diff
@@ -0,0 +1 @@
+from .transformation import *
```
Please add a newline at the end of the file

vyraun replied:
thanks, done

mmeyer39 commented on transformations/diacritic_removal/transformation.py:
```diff
@@ -0,0 +1,22 @@
+class DiacriticRemoval(SentenceOperation):
+
+    tasks = [TaskType.TEXT_CLASSIFICATION, TaskType.TEXT_TO_TEXT_GENERATION]
+    languages = ["en"]
```
Please add the `keywords` variable (see email from NL-Augmenter organizer).

vyraun replied:
Thanks, I have added the keywords now.

mmeyer39 commented:
This is a simple but effective transformation. 
It can help to rapidly clean datasets and input sentences so that they can be used for other purposes.

However an important limitation at the moment seems to be the use of this transformation for the English language. 
In English, the number of diacritic and accented characters is low.
The examples that you provided in the test file seems to reinforce this assertion.
Maybe it would also be useful to add support for languages where those characters are more common (e.g.: `French`, `Finish`, `Czech`).

It also feels like you need to have a good knowledge of your dataset in order to prevent meaning alteration as mentioned in the Readme's limitations. It might apply to broader domains than just chemistry or physics.

Overall, effective work!

vyraun commented:
> There is a typo in the PR name (tranformation -> transformation) and in the first commit message. Could you fix it?

Thanks, fixed the title.

vyraun commented:
> However an important limitation at the moment seems to be the use of this transformation for the English language.
> In English, the number of diacritic and accented characters is low.
> The examples that you provided in the test file seems to reinforce this assertion.
> Maybe it would also be useful to add support for languages where those characters are more common (e.g.: `French`, `Finish`, `Czech`).

Yes, there is no language-specific code in the transformation, so it can be applied to all languages. Where could I get the list of the language codes we are using in this repo? I think its ISO 639-1 code, but just wanted to confirm (https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes).

vyraun commented:
> Maybe it would also be useful to add support for languages where those characters are more common (e.g.: `French`, `Finish`, `Czech`).

Thanks, I have added extended the list of languages now. The code is actually language-agnostic, so languages with heavy use of  diacritics such as Czech are supported as well.

vyraun commented:
Thanks @mmeyer39 for the detailed review! I have addressed the issues through the commits above. Please let mw know if any change would be helpful or if the PR is good to go. Thanks!

kaustubhdhole commented:
Thanks @vyraun for these changes. I think transformations like these contribute towards making NL-Augmenter more kleen-star. I am merging this now. But please add the new keywords to the documentation file too in a separate PR. I will merge this one.
